### PR TITLE
process all new ROM dumps sent by Team Europe and Sean Riddle [David Haywood]

### PR DESCRIPTION
**Pull request by mamehaze**
new NOT WORKING machines
--------------
Interactive M.A.G. Motion Activated Gear: Toy Story and Beyond! Buzz Lightyear Galactic Adventure [David Haywood, Sean Riddle]
Interactive M.A.G. Motion Activated Gear: The Batman - Villains of Gotham City [David Haywood, Sean Riddle]
Interactive M.A.G. Motion Activated Gear: Spider-Man - Triple Threat [David Haywood, TeamEurope]
Who Wants to Be a Millionaire? (Play Vision, Plug and Play, UK, 8-bit version) [David Haywood, TeamEurope]
Excite Tennis (Japan) [David Haywood, TeamEurope]
Sudoku: Do You Sudoku? [David Haywood, TeamEurope]

new NOT WORKING clones
---------
Fun 2 Learn Smart Fit Park (France) [David Haywood, TeamEurope]

### CODE REVIEW

**Comment by mamehaze** [comment]
this used an unusual 'square' ROM glob, which took a lot of figuring out the pin-out of for Sean.

current ROM data should be in the correct order.

there are likely MCUs in the wearable parts of this though, as they can make sounds etc. without the main unit being turned on (number of parts involved is unknown)

File: src/mame/tvgames/elan_eu3a14.cpp [hunk 1]
```diff
@@ -788,6 +822,19 @@ void elan_eu3a14_state::radica_eu3a14(machine_config &config)
...
+{
+	radica_eu3a14(config);
+	m_vid->set_default_spriteramaddr(0x04); // at 0x800
+}
+
+void elan_eu3a14_state::radica_eu3a14_altspritebase_bat(machine_config& config)
+{
+	radica_eu3a14(config);
+	m_vid->set_default_spriteramaddr(0x0c); // at 0x1800
+}
```
[the remark this answers is not in the document]
Comment (mamehaze) [reply src/mame/tvgames/elan_eu3a14.cpp:835]: I'd rather find the register at some point, I thought I had it but I didn't.  Using a template just overcomplicates the code for something that, once we have enough evidence to figure out properly, will likely be simple.

So no, I think it's best as it is for now.

File: src/mame/tvgames/elan_eu3a14vid.cpp [hunk 2]
```diff
@@ -177,7 +178,7 @@ void elan_eu3a14vid_device::video_start()
 uint8_t elan_eu3a14vid_device::read_gfxdata(int offset, int x)
 {
 	address_space& fullbankspace = m_bank->space(AS_PROGRAM);
-	return fullbankspace.read_byte((offset+x) & 0x3fffff);
+	return fullbankspace.read_byte((offset+x) & 0x7fffff);
```
[the remark this answers is not in the document]
Comment (mamehaze) [reply src/mame/tvgames/elan_eu3a14vid.cpp:181]: performance really isn't an issue right now, all that has been changed here is the mask.  Again maybe once things work 100% it's worth considering, but for now it's not something I'd consider important to change.

**Comment by mamehaze** [comment]
I'm not really sure this can move forward more until further games are dumped using the hardware type , the existing set are not providing what I would call definitive evidence

**Comment by mamehaze** [comment]
(as always, for the hi-kara user carts somebody with knowledge of Japanese should add the tracks to the xml)

**Comment by cuavas** [comment]
Please split out the simple cartridge additions, and add photos and screenshots so that the titles and track lists can be sorted out before the stuff gets merged.  If it isn’t done beforehand, it won’t get done, and it will just add to the generally poor state of software lists.

**Comment by mamehaze** [comment]
I'd rather not split things up as it's just extra churn.  Photos were provided with the ROMs.

**Comment by mamehaze** [comment]
cart+pcb
cart+pcb
cart+pcb
cart+pcb
cart+pcb
cart+pcb
cart+pcb
cart+pcb
cart+pcb
cart+pcb

**Comment by mamehaze** [comment]
user10

**Comment by mamehaze** [comment]
user20

**Comment by mamehaze** [comment]
gc0012 and gc0013 on popira

**Comment by mamehaze** [comment]
Unwanted changes have been reverted.  The games no do nothing.

**Comment by mamehaze** [comment]
reverted the remaining changes, so this is now nothing but simple set additions.

**Comment by mamehaze** [comment]
I've removed all credits, as apparently credits are a problem.  Just credit [David Haywood, Team Europe, Sean Riddle] for everything.

**Comment by cuavas** [comment]
Your list of additions in the pull request description does not have credits for systems or software list item.  You did this last month as well resulting in TeamEurope’s cartridge dumps almost being credited to xtrasmiley.

I’ve got the cartridge additions ready to commit locally.  Who should be credited for each of the cartridge dumps?

**Comment by mamehaze** [comment]
The credits were originally there in the description, but I removed them in response to RB saying items couldn't have different credits in the same PR.  (Some of the MAG games were dumped by Sean and some were by Team Europe)

**Comment by mamehaze** [comment]
I really wish the team would make up their fucking minds, because I just feel like I'm jumping through hoops only to have to jump backwards through them again here.

**Comment by cuavas** [comment]
You’re misrepresenting what @rb6502 – he said that putting additions from different contributor in a single pull request is liable to cause confusion if don’t make it very clear what came from each contributor.

**Comment by mamehaze** [comment]
Well every item was individually credited, correctly, at the point he complained, how else am I meant to interpret it?

**Comment by cuavas** [comment]
> I really wish the team would make up their fucking minds, because I just feel like I'm jumping through hoops only to have to jump backwards through them again here.

No you aren’t:
* @rb6502 asked you not to mix additions from different contributors in the same pull request to avoid confusion.  You’ve done that again.
* I’ve asked you to provide images in pull requests for media where you can’t read the titles.  You still won’t do it without being prompted.
* I’ve asked you multiple times to use the wording for list headings in the whatsnew files.  You still won’t do that (e.g here you haven’t even separated working and non-working software list additions).
* You’ve been asked not to mix unrelated changes in the same pull request.  You keep doing that.

Once again, I have all the cartridge additions ready to commit.  Who did the each cartridge come from?

**Comment by mamehaze** [comment]
The request not to mix additions from different contributors is absolutely fucking ridiculous, the MAG stuff from Team Europe and Sean is in the same driver even.  Seems like a new rule is being made up every week.

@Osso13 I'm done with this, and MAME in general.  Any other ROMs I'll just be sending to you.

**Comment by cuavas** [comment]
> I really wish the team would make up their fucking minds, because I just feel like I'm jumping through hoops only to have to jump backwards through them again here.

You want to constantly accuse me of “rudeness” but you resort to swearing when asked for clarification.  Do you fail to see the irony?

> The request not to mix additions from different contributors is absolutely fucking ridiculous, the MAG stuff from Team Europe and Sean is in the same driver even. Seems like a new rule is being made up every week.

It isn’t a “rule”.  @rb6502 requested that you not mix them because you weren’t making it clear what came from where, leading to confusion and things almost being credited to the wrong people.  There’s less chance of that happening if the credits are the same for everything in a single PR.

**Comment by mamehaze** [comment]
As for where the cartridges came from, work it out yourself.  I've no further inclination to help you.

EVERYTHING was correctly credited before RB asked me to change it.

**Comment by cuavas** [comment]
> EVERYTHING was correctly credited before RB asked me to change it. EVERYTHING was clear and if you don't fucking believe me there's the edit history to prove it.

GitHub does not show edit history for comments or pull request descriptions.  We get a notification when a pull request is opened or a comment is added.  We do not get notifications when a pull request description or comment is edited.

The audit log shows the times when comments were edited.  However it doesn’t show the content before/after the edit.

The only edits where content before/after is visible are edits to pull request and issue titles.

There is no way I can see the description of the pull request at the time you claim it had proper credits.  Since it wasn’t there when you opened the pull request, it isn’t in the initial notification e-mail.  There is no user-visible edit history.

**Comment by cuavas** [comment]
Oh never mind, I found it.  It’s hidden behind JavaScript on the page itself.

**Comment by mamehaze** [comment]
The Softlist changes have been reverted.

**Comment by mamehaze** [comment]
So now you've stolen the Software List parts of this, when it would have been easier to just merge the whole thing, what happens to the rest of it?  Why is this simple case of adding some sets now such a nightmare?

Why is there so much needless bureaucracy here?

Who benefits from this?

**Comment by cuavas** [comment]
You know, it would have been a lot easier if you’d just said, “all the cartridges were from me and TeamEurope” or something.  You’re complaining about your workflow being interfered with, but at the same time going out of your way to make it difficult for everyone else.

**Comment by mamehaze** [comment]
I literally added the correct credits, and have been messed around.  You're pulling apart simple PRs, needlessly.  My mood would be better if you weren't making MAME such a fucking nightmare to work for.

**Comment by mamehaze** [comment]
and personally I'd consider blanking / ghosting me just because I didn't agree with your suggested changed before for over a month to be equally as rude as anything I've said.  I've reverted those changes now because I'd only made them to make sure the ROMs were good in the first place (which I thought had value, but I guess I should have made less effort)

if you're not going to give developers the freedom to develop, then development will not happen.  Micromanaging everything is not the way forward.

apparently you've done the same to that Sonic proto, just ghosted the PR, because you don't want to merge it.

absolutely shit project management.

**Comment by cuavas** [comment]
> So now you've stolen the Software List parts of this, when it would have been easier to just merge the whole thing, what happens to the rest of it? Why is this simple case of adding some sets now such a nightmare?

I “stole” the software list parts because I fixed the titles, added track lists for user cartrdiges, fixed the now-outdated comment about undumped main series e-kara cartridges, etc.  I also separated working and non-working items in the commit message so I don’t have discrepancies to resolve there when sorting out the release notes.

You weren’t going to fix the titles or add track lists yourself.  You don’t read Japanese, as you’re always so quick to remind us.  If I provided you with the details in review comments to add to your branch, you’d just complain that it’s inconvenient for you.

If I edited the files on your PR branch, it would add to the the already-convoluted commit history, and it would still be mixed up with the other unrelated stuff.

No matter what I do you’re going to complain.  At least this way the software list items go in cleanly, and it’s an software list-only change that doesn’t need to trigger macOS and Windows CI builds.

**Comment by cuavas** [comment]
> apparently you've done the same to that Sonic proto, just ghosted the PR, because you don't want to merge it.

No, it’s another case of a person mixing unrelated things together, and repeatedly adding to it.  Every time I go to try and salvage non-contentious parts from it, I find it’s changed again.

You seem to have this idea that if I haven’t actively commented on something in the last few days I’ve forgotten about it.  That isn’t true.

**Comment by mamehaze** [comment]
The prototype isn't contentious.

Holding a personal grudge at the expense of preserving something, just because you don't like that it was hacked (and you won't be seeing an unhacked copy) is a critical failure of MAME's neutrality.

**Comment by ICEknigh7** [comment]
> No, it’s another case of a person mixing unrelated things together, and repeatedly adding to it. Every time I go to try and salvage non-contentious parts from it, I find it’s changed again.

Of course a softlist PR will get changes after months of not being merged. Entries will keep getting added to it and additional mistakes will be fixed.

**Pull request merged**